Add attached files to report #26

mapeveri opened the pull request:
https://github.com/mozillahispano/mozbuzz/issues/26

Commit:
Merge pull request #61 from sebasmagri/master

Add WSGI script to replace wsgi XML configuration files

sebasmagri committed:
Add MySQL-python to the project's requirements

sebasmagri committed:
Update README.mkd

mapeveri committed:
Add attached files to report #26

sebasmagri commented on mozbuzz/buzz/models.py:
```diff
@@ -188,6 +189,7 @@ class Mention(SoftDeletableModel):
     relevant_audience = models.BooleanField(default=False)
     update_rate = models.IntegerField(max_length=1, choices=UPDATE_RATE)
     remarks = models.TextField(null=True, blank=True)
+    upload_file = models.FileField(upload_to="files/", default="", validators=[valid_extension], null=True, blank=True)
```
@mapeveri por favor asegúrate de que el nuevo código sea compliant con [PEP8](http://legacy.python.org/dev/peps/pep-0008/). Esta línea se ve bastante larga.

sebasmagri replied:
@mapeveri de esta manera estamos permitiendo solo un archivo como adjunto a un `Mention`. Por favor vamos a realizar los cambios necesarios para que se puedan adjuntar varios archivos a un mismo `Mention`.

mapeveri replied:
Ok lo del PEP8. 

Con respecto a lo de los adjuntos es lo que habíamos quedado. Yo te mande un email justamente explicando eso y me dijiste que sonaba bien. Este es una parte del email:

"Te hago algunas consultas:

1) Solo es un archivo que puede subir no?.
2) Si hay un archivo y agrega otro, eliminó el anterior?.
3) El archivo como lo muestro?. Simplemente pongo un icono para que lo descargue o muestro el contenido, si es una imagen es fácil, si es un pdf puedo poner un icono para descargar. No se...
"

Y vos me pusistes: "Tal como lo describes suena bien."

mapeveri replied:
"por favor asegúrate de que el nuevo código sea compliant con PEP8. Esta línea se ve bastante larga."

De esto ya hice el push, queda pendiente la respuesta de los varios adjuntos.

mapeveri replied:
@sebasmagri con respecto a los adjuntos hable con Berni y quedamos que por ahora soporte un solo archivo por mención. Dependiendo del uso que le de la gente, vemos si se agrega soporte a múltiples archivos o no.

sebasmagri commented on mozbuzz/buzz/models.py:
```diff
@@ -210,6 +212,11 @@ def getval(att):
+        if self.upload_file_clear:
+            self.upload_file = ""
+            self.upload_file_clear = False
+        super(Mention, self).save()'''
```
@mapeveri `'''foo'''` debería ser utilizado solo para documentación. En estos casos el viejo `#` es más adecuado. Sin embargo, trata de no enviar código deshabilitado por comentarios en los pull requests. En este caso es más conveniente sencillamente borrarlo.

mapeveri replied:
Esto esta de más. Me olvide de borrarlo, mil disculpas.

mapeveri replied:
Ya hice el push de esto también.

sebasmagri commented on mozbuzz/buzz/templates/buzz/mention.html:
```diff
@@ -1,13 +1,17 @@
+{% extends "base.html" %}
+
+{% load i18n query %}
+{% load files %}
```
@mapeveri coloquemos un `{% load ... %}` por línea acá.

mapeveri replied:
No entiendo a que te referís. No están cada uno en una linea?.

sebasmagri replied:
@mapeveri me refiero a hacer

```
{% load i18n %}
{% load query %}
{% load files %}
```

mapeveri replied:
@sebasmagri ya está hecho eso, fíjate si lo ves bien. Sinó decime que hago de nuevo el push.

sebasmagri replied:
@mapeveri acabé de verificar [en tu rama](https://github.com/mapeveri/mozbuzz/blob/4fac25a06c2013c6cd534a2fe9aa0f94d240dc41/mozbuzz/buzz/templates/buzz/mention.html) y lo veo igual. Por favor asegúrate de verificar en tu lado.

mapeveri replied:
Mira esta imagen como lo veo yo. Es la rama development de mi fork. Que puedo ser que este haciendo mal?. Gracias

![image](https://cloud.githubusercontent.com/assets/6276555/7554980/637acabc-f712-11e4-9cf8-2a5af408b157.png)

mapeveri replied:
Ah no te había entendido bien!. Yo veía así:

+{% load i18n query %}
+{% load files %}

Y decía "pero si esta en cada linea", pero no me di cuenta de que me pedías el i18n y query separado. Error mio, ya lo cambio!.

mapeveri replied:
@sebasmagri Listo ahora si!. Gracias

sebasmagri commented on mozbuzz/buzz/templates/buzz/mention_view.html:
```diff
@@ -1,4 +1,7 @@
-{% extends "base.html" %}{% load i18n query markup %}
+{% extends "base.html" %}
+
+{% load i18n query markup %}
```
Igualmente acá, un load por línea.

mapeveri replied:
Idem

sebasmagri commented on mozbuzz/buzz/templatetags/files.py:
```diff
@@ -0,0 +1,16 @@
+@register.filter(name='get_extension')
+@stringfilter
+def get_extension(value):
+    return os.path.splitext(value)[1]
```
Nada más démosle una pasada con `flake8` o algún similar para asegurarnos de seguir las guías de estilo.

mapeveri replied:
Dale genial

mapeveri replied:
Ya hice el push de esto también.

sebasmagri commented on mozbuzz/buzz/urls.py:
```diff
@@ -25,4 +27,4 @@
     url(r'^$', 'index', name="gracefully_degrade"),
 
-)
+) + static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT) 
```
Esto debería usarse si y solo si `settings.DEBUG == True`

mapeveri replied:
Ya hice el push de esto también.

sebasmagri commented:
Hola @mapeveri.

Muchísimas gracias por tu trabajo.

Te dejé algunos comentarios sobre los cambios. Por favor déjame saber si tienes alguna duda al respecto.

Cuando tengas los cambios, por favor hazle `squash` con los commits que ya subiste y actualiza el PR.

Saludos!

mapeveri commented:
@sebasmagri ahí deje varios comentarios.

Gracias

mapeveri committed:
Merge branch 'master' of https://github.com/mapeveri/mozbuzz into development

Conflicts:
	requirements.txt

mapeveri committed:
Add attached files to report - pull request

mapeveri commented:
@sebasmagri ya hice algunos cambios de lo que me pediste, queda que me respondas algunas preguntas que te hice para seguir.

Muchas gracias

mapeveri committed:
Add attached files to report #26

mapeveri committed:
Add attached files to report #26

mapeveri committed:
Add attached files to report #26

mapeveri committed:
Add attached files to report #26

mapeveri commented:
@sebasmagri decime si esta ok todo, o falta algo más para dar por finalizado este issue.

Gracias!

mapeveri committed:
Issue 26

sebasmagri commented:
Bien @mapeveri. Puedes hacer squash de todos los commits del Pull Request en uno solo?

Saludos.

mapeveri committed:
Issue 26 Add attached files to report

Add attached files to report - pull request

Add attached files to report #26

Add attached files to report #26

Add attached files to report #26

Add attached files to report #26

Issue 26

mapeveri commented:
@sebasmagri creo que ahí estaría. Sinó es así, me dirías cual es el comando exactamente. Gracias

sebasmagri commented:
@mapeveri dale una revisada a [git rebase -i](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html).

mapeveri commented:
@sebasmagri genial ahí me dio el mensaje "Successfully rebased and updated refs/heads/development." Así que creo que estaría ok.

mapeveri committed:
Issue 26

mapeveri committed:
# This is a combination of 7 commits.
# The first commit's message is:
Issue 26

# This is the 2nd commit message:

Issue 26

# This is the 3rd commit message:

Add attached files to report #26

# This is the 4th commit message:

Add attached files to report #26

# This is the 5th commit message:

Add attached files to report #26

# This is the 6th commit message:

Add attached files to report #26

# This is the 7th commit message:

Add attached files to report - pull request

mapeveri committed:
Issue 26

mapeveri committed:
Merge branch 'development' of https://github.com/mapeveri/mozbuzz into development
